pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -346,3 +362,96 @@ func validateRolloutStrategy(rolloutStrategy placementv1beta1.RolloutStrategy) e
 
 	return apiErrors.NewAggregate(allErr)
 }
+
+// validatePropertySelector validates the property selector
+func validatePropertySelector(propertySelector *placementv1beta1.PropertySelector) error {
+	return validatePropertySelectorRequirements(propertySelector.MatchExpressions)
+}
+
+func validatePropertySelectorRequirements(propertySelectorRequirements []placementv1beta1.PropertySelectorRequirement) error {
+	var allErr []error
+	for i, req := range propertySelectorRequirements {
+		if err := validateName(req.Name); err != nil {
+			allErr = append(allErr, fmt.Errorf("invalid property name %s: %w", req.Name, err))
+		}
+		if err := validateOperator(req.Operator, req.Values); err != nil {
+			allErr = append(allErr, err)
+		}
+		if err := validateValues(req.Values); err != nil {
+			allErr = append(allErr, fmt.Errorf("invalid values for property %s: %w", req.Name, err))
+		}
+		if err := checkContradictions(i, req, propertySelectorRequirements); err != nil {
+			allErr = append(allErr, err)
+		}
+	}
+	if len(allErr) > 0 {
+		return apiErrors.NewAggregate(allErr)
+	}
+	return nil
+}
+
+func validateName(name string) error {
+	if err := validation.IsQualifiedName(name); err != nil {
+		return fmt.Errorf("name is not a valid Kubernetes label name: %v", err)
+	}
+	return nil
+}
+
+func validateOperator(op placementv1beta1.PropertySelectorOperator, values []string) error {
```

Comment:
Hi Britania! The operator and value validation logic might be bundled -> for `eq`, there should be one value in quantity; I do not mind using this structure for now though. You could add a TODO comment as a hint for future re-structuring.